### YamlMime:ManagedReference
items:
- uid: System.IO.DriveNotFoundException
  id: DriveNotFoundException
  children:
  - System.IO.DriveNotFoundException.#ctor
  - System.IO.DriveNotFoundException.#ctor(System.String)
  - System.IO.DriveNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.IO.DriveNotFoundException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: DriveNotFoundException
  nameWithType: DriveNotFoundException
  fullName: System.IO.DriveNotFoundException
  type: Class
  summary: "Исключение возникает при попытке доступа к диску или данным совместного использования недоступен."
  remarks: "Класс DriveNotFoundException использует HRESULT COR_E_DIRECTORYNOTFOUND. Это исключение может вызываться при возникновении ошибки Win32 ERROR_PATH_NOT_FOUND HRESULT."
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class DriveNotFoundException : System.IO.IOException
  inheritance:
  - System.IO.IOException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.IO.DriveNotFoundException.#ctor
  id: '#ctor'
  parent: System.IO.DriveNotFoundException
  langs:
  - csharp
  name: DriveNotFoundException()
  nameWithType: DriveNotFoundException.DriveNotFoundException()
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.IO
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.IO.DriveNotFoundException&quot;> </xref> класса со строкой сообщения для системного сообщения и значением HRESULT, равным COR_E_DIRECTORYNOTFOUND."
  remarks: "Этот конструктор инициализирует <xref:System.Exception.Message%2A?displayProperty=fullName>свойство нового экземпляра системным сообщением, который описывает ошибку, например «Не удалось найти указанный каталог.»</xref:System.Exception.Message%2A?displayProperty=fullName> Это сообщение локализуется на текущую культуру системы.       <xref:System.Exception.InnerException%2A?displayProperty=fullName>Свойств нового экземпляра инициализируется `null`.</xref:System.Exception.InnerException%2A?displayProperty=fullName>"
  syntax:
    content: public DriveNotFoundException ();
    parameters: []
  overload: System.IO.DriveNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.IO.DriveNotFoundException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.IO.DriveNotFoundException
  langs:
  - csharp
  name: DriveNotFoundException(String)
  nameWithType: DriveNotFoundException.DriveNotFoundException(String)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.IO
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.IO.DriveNotFoundException&quot;> </xref> класс с указанной строкой сообщения и HRESULT, равным COR_E_DIRECTORYNOTFOUND."
  remarks: "Этот конструктор инициализирует <xref:System.Exception.Message%2A?displayProperty=fullName>нового экземпляра, используя свойство `message` параметр.</xref:System.Exception.Message%2A?displayProperty=fullName>       <xref:System.Exception.InnerException%2A?displayProperty=fullName>Свойств нового экземпляра инициализируется `null`.</xref:System.Exception.InnerException%2A?displayProperty=fullName>"
  syntax:
    content: public DriveNotFoundException (string message);
    parameters:
    - id: message
      type: System.String
      description: "Объект <xref:System.String>, описывающий ошибку.</xref:System.String> Код, вызывающий этот конструктор, должен убедиться, что эта строка локализована для текущего языка и региональных параметров системы."
  overload: System.IO.DriveNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.IO.DriveNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.IO.DriveNotFoundException
  langs:
  - csharp
  name: DriveNotFoundException(SerializationInfo,StreamingContext)
  nameWithType: DriveNotFoundException.DriveNotFoundException(SerializationInfo,StreamingContext)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.IO
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.IO.DriveNotFoundException&quot;> </xref> класса с указанными сведениями сериализации и контексте."
  syntax:
    content: protected DriveNotFoundException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "Объект <xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref> объект, содержащий сериализованные данные объекта о вызываемом исключении."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "Объект <xref:System.Runtime.Serialization.StreamingContext>, содержащий контекстные сведения об источнике или назначении исключения.</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.IO.DriveNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.IO.DriveNotFoundException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.IO.DriveNotFoundException
  langs:
  - csharp
  name: DriveNotFoundException(String,Exception)
  nameWithType: DriveNotFoundException.DriveNotFoundException(String,Exception)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.IO
  summary: "Инициализирует новый экземпляр <xref href=&quot;System.IO.DriveNotFoundException&quot;> </xref> класса с указанным сообщением об ошибке и ссылкой на внутреннее исключение, вызвавшее данное исключение."
  remarks: "Исключение, созданное как прямой результат предыдущего исключения, должно содержать ссылку на предыдущее исключение в <xref:System.Exception.InnerException%2A>свойство.</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>Свойство возвращает то же значение, которое передается конструктору, или `null` Если <xref:System.Exception.InnerException%2A>свойства не предоставляет конструктору значение внутреннего исключения.</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       В следующей таблице показаны исходные значения свойств для экземпляра класса <xref:System.IO.DirectoryNotFoundException>.</xref:System.IO.DirectoryNotFoundException>      | Свойство | Значение |   |--------------|-----------|   | <xref:System.Exception.InnerException%2A>| Ссылка на внутреннее исключение. |   | <xref:System.Exception.Message%2A>| Строка сообщения об ошибке. |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public DriveNotFoundException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "Сообщение об ошибке с объяснением причин исключения."
    - id: innerException
      type: System.Exception
      description: "Исключение, которое стало причиной текущего исключения. Если `innerException` не <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, текущее исключение сгенерировано в <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> блок, обрабатывающем внутреннее исключение."
  overload: System.IO.DriveNotFoundException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.IO.IOException
  isExternal: false
  name: System.IO.IOException
- uid: System.IO.DriveNotFoundException.#ctor
  parent: System.IO.DriveNotFoundException
  isExternal: false
  name: DriveNotFoundException()
  nameWithType: DriveNotFoundException.DriveNotFoundException()
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException()
- uid: System.IO.DriveNotFoundException.#ctor(System.String)
  parent: System.IO.DriveNotFoundException
  isExternal: false
  name: DriveNotFoundException(String)
  nameWithType: DriveNotFoundException.DriveNotFoundException(String)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.IO.DriveNotFoundException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.IO.DriveNotFoundException
  isExternal: false
  name: DriveNotFoundException(SerializationInfo,StreamingContext)
  nameWithType: DriveNotFoundException.DriveNotFoundException(SerializationInfo,StreamingContext)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.IO.DriveNotFoundException.#ctor(System.String,System.Exception)
  parent: System.IO.DriveNotFoundException
  isExternal: false
  name: DriveNotFoundException(String,Exception)
  nameWithType: DriveNotFoundException.DriveNotFoundException(String,Exception)
  fullName: System.IO.DriveNotFoundException.DriveNotFoundException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.IO.DriveNotFoundException.#ctor*
  parent: System.IO.DriveNotFoundException
  isExternal: false
  name: DriveNotFoundException
  nameWithType: DriveNotFoundException.DriveNotFoundException
